### ISSUE TITLE
In-App Purchase Ads (Interstitial)

### ISSUE THREAD

**Issue by seal-ua** [googleads/googleads-mobile-unity#143]
Hi everyone. 
We have some problems with In-App Purchase Ads.

We are using Unity 5.1.2f1 with Google Mobile Ads Unity Plugin v2.3.1.
We tried to use GoogleMobileAdsDemoScript.cs as test code (AndroidPublicKey and Interstitial adUnitId were replaced to ours). 
It shows In-App Purchase Ads correctly, but after click it nothing happens.

In logs we can see this errors:
12-08 17:54:06.050: W/Ads(31999): InAppPurchaseListener is not set. Try to launch default purchase flow.
12-08 17:54:06.050: W/Ads(31999): PlayStorePurchaseListener is not set.
12-08 17:54:06.100: W/Ads(31999): Received non-success response code 302 from pinging URL: http://googleads.g.doubleclick.net/aclk?sa……

Any ideas where problem can be?

**Comment by ericleich** [comment]
Hello,

Based on the logs, I'm guessing you didn't do step 3 from [these directions](https://developers.google.com/admob/android/games#in-app_purchase_ads):

    interstitial.SetInAppPurchaseHandler(...);

Calling this will set a PlayStorePurchaseListener under the hood, which should lift the warning, and allow the SDK to continue.

**Comment by bcjtools** [comment]
I think the problem here is that documentation is rather weak with regards to the In App Purchase section.  I had no problem implementing ads using the plugin (might have helped that I implemented AdMob ads in two apps a couple years ago) but exactly how to implement the In App Purchase is a little confusing.  The demo doesn't even have a non-working version.  I also looked at the implementation on Android itself and there are two buttons which with no click listener on them, so one would wonder how that might work?  What's missing is exactly what you call if a users clicks a button to purchase.

The above suggestion is shown in the HelloWorld demo but easily missed.  Seems from reading here and on the group people spend too much time trying to figure it out when just a few more lines of example code would help. Note too that the directions on the admob site don't necessarily coalesce with the way things are done in the Unity plugin.

So is there a more complete demo found somewhere?
